data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/source/oteltrace/OTelTraceGrpcService.java

```diff
@@ -7,46 +7,89 @@
+import java.util.List;
 import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
 
 public class OTelTraceGrpcService extends TraceServiceGrpc.TraceServiceImplBase {
     private static final Logger LOG = LoggerFactory.getLogger(OTelTraceGrpcService.class);
 
     public static final String REQUEST_TIMEOUTS = "requestTimeouts";
     public static final String REQUESTS_RECEIVED = "requestsReceived";
+    public static final String BAD_REQUESTS = "badRequests";
+    public static final String REQUESTS_TOO_LARGE = "requestsTooLarge";
+    public static final String INTERNAL_SERVER_ERROR = "internalServerError";
+    public static final String SUCCESS_REQUESTS = "successRequests";
+    public static final String PAYLOAD_SIZE = "payloadSize";
+    public static final String REQUEST_PROCESS_DURATION = "requestProcessDuration";
 
     private final int bufferWriteTimeoutInMillis;
-    private final Buffer<Record<ExportTraceServiceRequest>> buffer;
+    private final RecordType recordType;
+    private final OTelProtoCodec.OTelProtoDecoder oTelProtoDecoder;
+    private final Buffer<Record<Object>> buffer;
 
     private final Counter requestTimeoutCounter;
     private final Counter requestsReceivedCounter;
+    private final Counter successRequestsCounter;
+    private final Counter badRequestsCounter;
+    private final Counter requestsTooLargeCounter;
+    private final Counter internalServerErrorCounter;
+    private final DistributionSummary payloadSizeSummary;
+    private final Timer requestProcessDuration;
 
 
     public OTelTraceGrpcService(int bufferWriteTimeoutInMillis,
-                                Buffer<Record<ExportTraceServiceRequest>> buffer,
+                                final RecordType recordType,
+                                final OTelProtoCodec.OTelProtoDecoder oTelProtoDecoder,
+                                final Buffer<Record<Object>> buffer,
                                 final PluginMetrics pluginMetrics) {
         this.bufferWriteTimeoutInMillis = bufferWriteTimeoutInMillis;
+        this.recordType = recordType;
         this.buffer = buffer;
+        this.oTelProtoDecoder = oTelProtoDecoder;
 
         requestTimeoutCounter = pluginMetrics.counter(REQUEST_TIMEOUTS);
         requestsReceivedCounter = pluginMetrics.counter(REQUESTS_RECEIVED);
+        badRequestsCounter = pluginMetrics.counter(BAD_REQUESTS);
+        requestsTooLargeCounter = pluginMetrics.counter(REQUESTS_TOO_LARGE);
+        internalServerErrorCounter = pluginMetrics.counter(INTERNAL_SERVER_ERROR);
+        successRequestsCounter = pluginMetrics.counter(SUCCESS_REQUESTS);
+        payloadSizeSummary = pluginMetrics.summary(PAYLOAD_SIZE);
+        requestProcessDuration = pluginMetrics.timer(REQUEST_PROCESS_DURATION);
     }
 
 
     @Override
     public void export(ExportTraceServiceRequest request, StreamObserver<ExportTraceServiceResponse> responseObserver) {
+        if (recordType.equals(RecordType.otlp)) {
```

Comment:
You can use `==` for enums in Java. And it is often preferred because it is null-safe.